test/test_uncertainty.py

```diff
@@ -100,40 +103,50 @@ def test_daily_uncertainty(get_config: Fixture[Callable[[str], dict]],
-        Id of a basin for which to check the results
+        The ID of the basin for which predictions are being checked.
+    negative_sample_handling : str
+        Strategy used to handle negative samples during training ("truncate" or "clip"),
+        which determines whether non-negativity is strictly enforced in the output.
     """
-    results = _get_basin_results(config.run_dir, 1)[basin]['1D']['xr'].isel(time_step=-1)
+    results = get_basin_results(config.run_dir, 1)[basin]['1D']['xr'].isel(time_step=-1)
     
-    print("\n[DEBUG] Available variables:", results.data_vars)
     sample_key = f"{config.target_variables[0]}_sim"
-    assert sample_key in results.data_vars
-    assert "samples" in results[sample_key].dims # evaluation produces a samples dimension (probabilistic output)
-    assert not pd.isna(results[sample_key]).any()  # Check for NaN values in the samples
+    assert sample_key in results.data_vars, f"Expected {sample_key} in results, got {results.data_vars}"
+    # The model evaluation should produce a 'samples' dimension (probabilistic output)
+    assert "samples" in results[sample_key].dims, f'"samples" dimension not found in {sample_key}' 
 
-    # get the test date range from the config
-    test_start_date, test_end_date = _get_test_start_end_dates(config)
-    # check that samples in the test period are not NaN
-    test_dates = pd.date_range(test_start_date, test_end_date, freq='D')
-    test_vals = results.sel(date=test_dates)
-    assert not pd.isna(test_vals[sample_key]).any()  # Check for NaN values in the test period
+    # Assert the number of samples in the output matches the config
+    assert results[sample_key].shape[1] == config.n_samples, f'Expected {config.n_samples} samples, got {results[sample_key].shape[0]}'
 
-    if negative_sample_handling == 'truncate':
-        # check that no samples are negative
-        assert (test_vals[sample_key] >= 0).all()
-    elif negative_sample_handling == 'clip':
-        # check that no samples are negative
-        assert (test_vals[sample_key] >= 0).all()
+    # Check that the results file has the correct date range
+    test_start_date, test_end_date = get_test_start_end_dates(config)
+    assert pd.to_datetime(results['date'].values[0]) == test_start_date.floor('D')
+    assert pd.to_datetime(results['date'].values[-1]) == test_end_date.floor('D')
 
+    # Check that no NaN values are present in the generated samples 
+    test_dates = pd.date_range(test_start_date, test_end_date, freq='D')
+    test_vals = results.sel(date=test_dates)
+    # Assert all sample values are finite
+    assert np.isfinite(test_vals[sample_key].values).all(), f'Found non-finite values in {sample_key}'
+
+    if negative_sample_handling in ["truncate", "clip"]:
+        # For 'truncate' or 'clip', we expect all non-negative values
+        negative_vals = test_vals[sample_key].values[test_vals[sample_key].values < 0]
+        assert np.allclose(negative_vals, 0.0), (
```

Comment:
seems like we need to be a little more lenient here and increase the tolerance in the allclose call (from the logs the values are something like -6e-8).